src/main/java/com/iota/iri/network/pipeline/BroadcastStage.java

```diff
@@ -54,6 +58,15 @@ public ProcessingContext process(ProcessingContext ctx) {
             }
         }
 
+        // Check the transaction solidifier to see if there are solid transactions that need to be broadcast.
+        // If so, forward them to the BroadcastStageQueue to be processed.
+        TransactionViewModel transactionToBroadcast;
+        if((transactionToBroadcast = transactionSolidifier.getNextTxInBroadcastQueue()) != null){
+            ctx.setNextStage(TransactionProcessingPipeline.Stage.BROADCAST);
+            ctx.setPayload(new BroadcastPayload(payload.getOriginNeighbor(), transactionToBroadcast));
+            return ctx;
+        }
```

Comment:
hmm I thought you were going to do a `while` loop

I guess what you did here works as well
Not sure about performance what is better?

Did you test?

Reply:
I did test it in ICC and there were no issues with this approach. Originally I was going to do a while, but I was worried about holding up the forwarding of received transactions. With this approach it will mesh the received and the solidified transactions in the queue for the broadcasts.